dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs

```diff
@@ -693,7 +693,18 @@ internal ClientCore(ILogger? logger = null)
                             OpenAIFunctionToolCall.TrackStreamingToolingUpdate(update.ToolCallUpdate, ref toolCallIdsByIndex, ref functionNamesByIndex, ref functionArgumentBuildersByIndex);
                         }
 
-                        var openAIStreamingChatMessageContent = new OpenAIStreamingChatMessageContent(update, update.ChoiceIndex ?? 0, this.DeploymentOrModelName, metadata) { AuthorName = streamedName };
+                        AuthorRole? role = null;
+                        if (streamedRole.HasValue)
+                        {
+                            role = new AuthorRole(streamedRole.Value.ToString());
+                        }
+
+                        OpenAIStreamingChatMessageContent openAIStreamingChatMessageContent =
+                            new(update, update.ChoiceIndex ?? 0, this.DeploymentOrModelName, metadata)
+                            {
+                                AuthorName = streamedName,
+                                Role = role,
```

Comment:
From PR description:

> StreamingChatMessageContent.AuthorRole always null

I'm not sure it's true. I'm trying to run chat completion streaming example and I'm getting a role in the first chunk:
https://github.com/microsoft/semantic-kernel/blob/745e64a0265923c926c26a3e61127433f6506ba1/dotnet/samples/Concepts/ChatCompletion/OpenAI_ChatCompletionStreaming.cs#L146-L158

We don't set `Role` explicitly in `ClientCore.cs` in streaming scenario, but it is passed as part of `update` object into `OpenAIStreamingChatMessageContent`:
![image](https://github.com/microsoft/semantic-kernel/assets/13853051/6d85e567-f352-49e4-ad64-99ffb89317a0)

Then it is processed in constructor in this place:
https://github.com/microsoft/semantic-kernel/blob/8033515a86964a6b712886b0910d5b5fa8accfe7/dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/OpenAIStreamingChatMessageContent.cs#L30-L36

Could you please double-check your scenario? There could be the case when no changes are required for this functionality. Thanks!

Reply:
The `ModelId` and other properties are included with each chunk.  I'm not sure why we wouldn't desire parity with the role definition?

Reply:
Makes sense. The description `StreamingChatMessageContent.AuthorRole always null` confused me, it was not clear which scenario is fixed as part of this PR.